### DISCUSSION TITLE
Validation [when] rule

### DISCUSSION THREAD

**Discussion by infureal** [laravel/framework#46803] [Ideas]
For example I want to validate something when other field has specific value:
```php
'<key>' => 'when:<other_key>,<value>,<rule>'
```
For more specific example:
```php
'type' => ['required', 'in:email,phone']
'to' => ['required', 'when:type,email,email', 'when:type,phone,check_phone'], // check_phone is isolated validation rule, that can be used separately
```

In second example we have `to` field that validated specificly by `type` field. 

Now I can do it like this, but I think there is better way: 
```php
$toValidation = match ($data['type']) {
   'email' => 'email',
   'phone' => 'check_phone',
   default => 'string'
};

// then add $toValidation to rules
```

**Comment by henzeb** [comment]
I think you better use 2 fields, phone and email, and use required_if:type,email and required_if:type,phone

**Reply by joelharkes** [reply]
or email=> required_if,notphone
phone=> required_if,notEmail.

something like this?

**Reply by joelharkes** [reply]
i would suggest just splitting the forms and the API calls up, as probably the logic changes based on the input.
email you want to send an email, phone you want to send an sms?

**Reply by infureal** [reply]
I'm using Apache Kafka and validating incomming message. And I can't split it to two separated topics

**Comment by infureal** [comment]
Nevermind. I think it will me missunderstaning and generate a lot of questions. Sorry for bothering
